### ISSUE TITLE
Fail to sync ledger data in fabric-samples/test-network with fabric v2.1.1

### ISSUE THREAD

**Issue by nekia** [FujitsuLaboratories/Ledger-Data-Refiner#3]
Block/Tx data that generated after starting the dashboard seems to fail to be synced correctly.

I can see the following errors on refiner container every 5 seconds:
```
time="2021-01-29T11:26:51Z" level=error msg="failed to insert block into the database" block_id=7 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:51Z" level=error msg="failed to insert block into the database" block_id=8 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:51Z" level=error msg="failed to insert block into the database" block_id=9 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:51Z" level=error msg="failed to insert block into the database" block_id=10 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:51Z" level=error msg="failed to insert block into the database" block_id=6 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:56Z" level=error msg="failed to insert block into the database" block_id=6 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:56Z" level=error msg="failed to insert block into the database" block_id=7 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:56Z" level=error msg="failed to insert block into the database" block_id=8 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:56Z" level=error msg="failed to insert block into the database" block_id=9 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:26:56Z" level=error msg="failed to insert block into the database" block_id=10 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
```

```
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=6 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" lSevel=error msg="failed to insert block into the database" block_id=7 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=8 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=9 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=10 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=9 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=10 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=6 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=7 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
time="2021-01-29T11:28:36Z" level=error msg="failed to insert block into the database" block_id=8 error="failed to insert tx into the database: ERROR: invalid byte sequence for encoding \"UTF8\": 0x00 (SQLSTATE 22021)"
```

### Steps to recreate the issue
+ Start fabric-samples/test-network with fabric v2.1.1
+ Start Ledger Data Refiner as a container
+ Generate some transactions on the test-network

**Comment by nekia** [comment]
Please find attached logs for more detail of this issue:
* [refiner.log](https://github.com/FujitsuLaboratories/Ledger-Data-Refiner/files/5908744/refiner.log)
* [refinerdb.log](https://github.com/FujitsuLaboratories/Ledger-Data-Refiner/files/5908745/refinerdb.log)

**Comment by nekia** [comment]
We still have same issue even after applying the latest commit. 
[refiner_7ff6dfe.log](https://github.com/FujitsuLaboratories/Ledger-Data-Refiner/files/5938499/refiner_7ff6dfe.log)

**Comment by wjbbig** [comment]
It seems that in the fabric version 2.0~2.2 there is an invalid character in the read set, and it cannot be found directly from the log. Now we have updated the code, please run it again. @nekia

**Comment by nekia** [comment]
@wjbbig Thank you for update. I tried the latest code and confirmed I don't see the encoding error any more. But the dashboard shortly gets not responsive.
<img width="597" alt="2021-02-08_23h45_28" src="https://user-images.githubusercontent.com/799608/107221861-4f080200-6a68-11eb-8c65-34bbca08ba30.png">

Now I can see another errors in refiner log.
```shell
$ grep -i error refiner_dff4526.log
time="2021-02-08T12:43:39Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:43:44Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:43:49Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:43:54Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:43:59Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:44:04Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:44:09Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:44:14Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:44:19Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
time="2021-02-08T12:44:24Z" level=error msg="json: cannot unmarshal number into Go value of type map[string]interface {}"
```
[refiner_dff4526.log](https://github.com/FujitsuLaboratories/Ledger-Data-Refiner/files/5943759/refiner_dff4526.log)

**Comment by wjbbig** [comment]
Sorry, I forgot to recover the commented out code and did not find this problem yesterday. I found that the official API `json.Valid(string) bool` of golang checks "1" as JSON format, which caused the above problem.

**Comment by nekia** [comment]
@wjbbig Now I can see dashboard, history view and advanced query view. It seems to be working fine. Thanks.
If a new block is created on the fabric network, when can I expect seeing the update on the dashboard? As far as I checked, I can't see newly generated block on the dashboard even after reloading browser.

**Comment by nekia** [comment]
With the fix d7e8a1d665f9a583742a07d3f108635aa39f6d3b, now I can see the update of newly generated blocks on the dashboard as I expect. Thanks @wjbbig
